docs/functionality/motors.md

```diff
@@ -0,0 +1,135 @@
+# Motors interface under tfc
+## SpeedRatio
+A concept that
+maps a motors capable
+range from 1-100%
+
+### Stopping
+ (-1<=>1) is outside the motors
+ range by definition and
+ the motor shall be stopped
+ if a speed is applied in that range.
+
+### Rationalization
+We have a plethora of
+use cases where precise motor
+speeds are not essential.
+We also utilize a large range
+of frequencies, rpm's and gear
+ratios in our motors, causing
+many other physical quantities
+to be a bad fit for simple general
+purpose use cases.
+
+## Activating a running motor
+If activating a running motor
+the motor shall run with the new
+parameters provided, fe. changing
+speeds. The new speed shall be used.
+
+This also applies if the motor
+is switched from running continuously 
+to running with a set end point. The
+motor shall than stop after the distance
+or time provided.
+
+
+## Run modes
+Each motor interface can decide
+its basic capability aside from 
+the basic run mode at a fractional
+speed. If the motor interface cannot
+complete it's given task a std::error_code
+shall be returned with an enum and string as to
+why.
+
+The run modes can be thought of
+as 
+- just run
+- just run for some time
+- just run for some amount
+- just go there
+
+These are then hidden behind different apis
+to facilitate different use cases.
+f.e. 
+a pump uses the `.pump` function call
+while a conveyor uses the `.convey` function.
+
+### Why complicate simple motors
+Many different motor have different properties
+Servos and synchronous motors have positions.
+But a async motor with a nominal mm/s @ 50hz
+also has a decent distance estimate, an 
+async motor coupled with a tachometer or an
+encoder has even better precision.
+
+
+
+## Interface idea
+```cpp
+int main(){
+    motor_types m{};
```

Comment:
```suggestion
    motor_interface m{};
```
?

Reply:
Could be less confusing
general_interface
pump_interface : general_interface
convey_interface : general_interface
rotate_interface : general_interface